shell/browser/api/electron_api_utility_process.cc

```diff
@@ -274,116 +274,103 @@ gin::WrapperInfo UtilityProcessWrapper::kWrapperInfo = {
+#if BUILDFLAG(IS_WIN)
+      HANDLE read = nullptr;
+      HANDLE write = nullptr;
+      if (!::CreatePipe(&read, &write, nullptr, 0)) {
+        PLOG(ERROR) << "pipe creation failed";
+        return;
+      }
+#elif BUILDFLAG(IS_POSIX)
+      int pipe_fd[2];
+      if (HANDLE_EINTR(pipe(pipe_fd)) < 0) {
+        PLOG(ERROR) << "pipe creation failed";
+        return;
+      }
+#endif
+      if (handle == IOHandle::STDOUT) {
+#if BUILDFLAG(IS_WIN)
+        stdout_write.Set(write);
+        stdout_reader_ = std::make_unique<StdoutPipeReader>(
+            weak_factory_.GetWeakPtr(), read);
+#elif BUILDFLAG(IS_POSIX)
+        fds_to_remap.push_back(std::make_pair(pipe_fd[1], STDOUT_FILENO));
+        stdout_reader_ = std::make_unique<StdoutPipeReader>(
+            weak_factory_.GetWeakPtr(), pipe_fd[0]);
+#endif
+        if (!stdout_reader_->Start()) {
+          PLOG(ERROR) << "stdout output watcher thread failed to start.";
+          PipeIOBase::Shutdown(std::move(stdout_reader_));
+#if BUILDFLAG(IS_WIN)
+          CloseHandle(write);
+#elif BUILDFLAG(IS_POSIX)
+          IGNORE_EINTR(close(pipe_fd[1]));
+#endif
+          return;
+        }
+      } else if (handle == IOHandle::STDERR) {
+#if BUILDFLAG(IS_WIN)
+        stderr_write.Set(write);
+        stderr_reader_ = std::make_unique<StderrPipeReader>(
+            weak_factory_.GetWeakPtr(), read);
+#elif BUILDFLAG(IS_POSIX)
+        fds_to_remap.push_back(std::make_pair(pipe_fd[1], STDERR_FILENO));
+        stderr_reader_ = std::make_unique<StderrPipeReader>(
+            weak_factory_.GetWeakPtr(), pipe_fd[0]);
+#endif
+        if (!stderr_reader_->Start()) {
+          PLOG(ERROR) << "stderr output watcher thread failed to start.";
+          PipeIOBase::Shutdown(std::move(stderr_reader_));
+#if BUILDFLAG(IS_WIN)
+          CloseHandle(write);
+#elif BUILDFLAG(IS_POSIX)
+          IGNORE_EINTR(close(pipe_fd[1]));
+#endif
+          return;
+        }
+      }
+    } else if (type == IOType::IGNORE) {
+#if BUILDFLAG(IS_WIN)
+      HANDLE handle =
+          CreateFileW(L"NUL", FILE_GENERIC_WRITE | FILE_READ_ATTRIBUTES,
+                      FILE_SHARE_READ | FILE_SHARE_WRITE, nullptr,
+                      OPEN_EXISTING, 0, nullptr);
+      if (handle == INVALID_HANDLE_VALUE) {
+        PLOG(ERROR) << "stdout failed to create null handle";
+        return;
+      }
+#elif BUILDFLAG(IS_POSIX)
+      int devnull = open("/dev/null", O_WRONLY);
+      if (devnull < 0) {
+        PLOG(ERROR) << "failed to open /dev/null";
+        return;
+      }
+#endif
+      if (handle == IOHandle::STDOUT) {
+#if BUILDFLAG(IS_WIN)
+        stdout_write.Set(handle);
+#elif BUILDFLAG(IS_POSIX)
+        fds_to_remap.push_back(std::make_pair(devnull, STDOUT_FILENO));
+#endif
+      } else if (handle == IOHandle::STDERR) {
+#if BUILDFLAG(IS_WIN)
+        stderr_write.Set(handle);
+#elif BUILDFLAG(IS_POSIX)
+        fds_to_remap.push_back(std::make_pair(devnull, STDERR_FILENO));
+#endif
+      }
```

Comment:
I feel like this would be easier to read with one big #if / #elif, rather than a bunch of little ones, even though the structure would be repeated.

Reply:
https://github.com/electron/electron/pull/34980/commits/1b3ac3f62612fa286f265e8ecc191149c9c65ce7